SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -0,0 +1,505 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSStorage.h"
+#import "AppContext.h"
+#import "NSData+Base64.h"
+#import "TSAttachmentStream.h"
+#import "TSStorageManager.h"
+#import <Curve25519Kit/Randomness.h>
+#import <SAMKeychain/SAMKeychain.h>
+
+NS_ASSUME_NONNULL_BEGIN
```

Comment:
Much of this code was pulled verbatim from `TSStorageManager`.  I added the concept of a db connection delegate, which will become more important soon.